bot/internal/bot/bot.go

```diff
@@ -153,6 +159,42 @@ func classifyChanges(e *env.Environment, files []github.PullRequestFile) env.Cha
 	return ch
 }
 
+// approverCount returns the number of required approvers for the PR by comparing the files included
+// in the PR against a set of paths that only require a single approver and the PR author against
+// a set of authors that only require a single approver. 1 is returned when all of the
+// files match a single approver path or the PR author matches one of the authors, otherwise
+// env.DefaultApproverCount is returned.
+func approverCount(authors, paths []string, author string, files []github.PullRequestFile) int {
```

Comment:
Should we enforce some sort of naming convention for bots that only require a single approval as an extra precaution? I don't think we'd ever want an employee to unconditionally only require a single approval.

Alternatively, we can just require that any author configured as a single approval also passes `isAllowedRobot`.

Reply:
Good idea. I added a test to ensure only bots can be configured as single approver authors. We can also export `isAllowedRobot` from the `review` package and test in `approverCount` if you think we should have a secondary set of checks.